### PR TITLE
fastjet: Adding thread safety variant

### PR DESCRIPTION
**Pull request by kjvbrt**
Adds variant to by default enable limited thread safety. Limited thread safety should not hinder performance. See http://fastjet.fr/repo/fastjet-doc-3.4.2.pdf

### CODE REVIEW

File: var/spack/repos/builtin/packages/fastjet/package.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ class Fastjet(AutotoolsPackage):
 
     variant("shared", default=True, description="Builds a shared version of the library")
     variant("auto-ptr", default=False, description="Use auto_ptr")
+    variant("limited-thread-safety", default=True, description="Enables limited thread safety")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/fastjet/package.py]: Is this limited compared to full thread safety or limited compared to no thread safety at all? The description doesn't make that clear.

Comment (wdconinc) [reply var/spack/repos/builtin/packages/fastjet/package.py]: There also appears to be a `--thread-safety` option too which would be useful to expose here too. Can this be a multi value with none, limited, full?

Comment (kjvbrt) [reply var/spack/repos/builtin/packages/fastjet/package.py]: `--enable-thread-safety` enables full thread safety and `--enable-limited-thread-safety` enables limited thread safety compared to full thread safety.

**Review by vvolkl — APPROVED** [review]

File: var/spack/repos/builtin/packages/fastjet/package.py [hunk 2]
```diff
@@ -61,6 +61,15 @@ class Fastjet(AutotoolsPackage):
...
     variant("shared", default=True, description="Builds a shared version of the library")
     variant("auto-ptr", default=False, description="Use auto_ptr")
+    variant(
+        "limited-thread-safety",
+        default=True,
+        when="@3.4.0:",
+        description="Enables limited thread safety",
+    )
+    variant(
+        "thread-safety", default=False, when="@3.4.0:", description="Enables full thread safety"
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/fastjet/package.py]: What happens if both variants are set? Does this need a `conflict`?

Comment (kjvbrt) [reply var/spack/repos/builtin/packages/fastjet/package.py]: In that case the full thread safety is enabled.

Comment (wdconinc) [reply var/spack/repos/builtin/packages/fastjet/package.py]: What I mean is that these two binary variants set up four possible states, but control a three-state system. That's not ideal, so we need to disallow one state or use a three-valued variant.

Comment (kjvbrt) [reply var/spack/repos/builtin/packages/fastjet/package.py]: I see, now there is multi-valued variant with three possible values "none", "limited", "full".

**Review by wdconinc — APPROVED** [review]
Looks good!

**Pull request merged**